nikola/post.py

```diff
@@ -754,24 +762,32 @@ def remaining_paragraph_count(self):
     def source_link(self, lang=None):
         """Return absolute link to the post's source."""
         ext = self.source_ext(True)
-        link = "/" + self.destination_path(lang=lang, extension=ext, sep='/')
+        link = "/" + self.destination_path(lang=lang, extension=ext, sep='/', _force_source=True)
         link = utils.encodelink(link)
         return link
 
-    def destination_path(self, lang=None, extension='.html', sep=os.sep):
+    def destination_path(self, lang=None, extension='.html', sep=os.sep, _force_source=False):
         """Destination path for this post, relative to output/.
 
         If lang is not specified, it's the current language.
         Extension is used in the path if specified.
         """
         if lang is None:
             lang = nikola.utils.LocaleBorg().current_lang
+        if _force_source:
+            folder = self.folder
```

Comment:
Ah. I think I misunderstood something about  `Post.source_link`. Should be fixed now.